Promtail: close of closed channel

Hi, I'm running Promtail (as a DaemonSet) + Loki in a GKE cluster with multiple nodes. For some reason, only one of the Promtail pods is erroring and keeps getting restarted.

The pod log has this error in it:
```
panic: close of closed channel

goroutine 207 [running]:
github.com/grafana/loki/pkg/promtail.(*Positions).Stop(...)
	/go/src/github.com/grafana/loki/pkg/promtail/position.go:65
github.com/grafana/loki/pkg/promtail.(*Target).run.func1(0xc000d42de0)
	/go/src/github.com/grafana/loki/pkg/promtail/target.go:142 +0x13b
github.com/grafana/loki/pkg/promtail.(*Target).run(0xc000d42de0)
	/go/src/github.com/grafana/loki/pkg/promtail/target.go:197 +0x116e
created by github.com/grafana/loki/pkg/promtail.NewTarget
	/go/src/github.com/grafana/loki/pkg/promtail/target.go:125 +0x8b6
```

All the other pods seem to be running fine. It's worth noting that the Loki pod is also running on this particular node.

--- a/pkg/promtail/promtail.go
+++ b/pkg/promtail/promtail.go
@@ -52,5 +52,6 @@ func (p *Promtail) Run() error {
 func (p *Promtail) Shutdown() {
 	p.server.Shutdown()
 	p.targetManager.Stop()
+	p.positions.Stop()
 	p.client.Stop()
 }

--- a/pkg/promtail/target.go
+++ b/pkg/promtail/target.go
@@ -136,10 +136,9 @@ func (t *Target) run() {
 	defer func() {
 		helpers.LogError("closing watcher", t.watcher.Close)
 		for _, v := range t.tails {
+			helpers.LogError("updating tailer last position", v.markPosition)
 			helpers.LogError("stopping tailer", v.stop)
 		}
-		//Save positions
-		t.positions.Stop()
 		level.Debug(t.logger).Log("msg", "watcher closed, tailer stopped, positions saved")
 		close(t.done)
 	}()

--- a/pkg/promtail/target_test.go
+++ b/pkg/promtail/target_test.go
@@ -69,6 +69,7 @@ func TestLongSyncDelayStillSavesCorrectPosition(t *testing.T) {
 	}
 
 	target.Stop()
+	positions.Stop()
 
 	buf, err := ioutil.ReadFile(filepath.Clean(positionsFileName))
 	if err != nil {
@@ -161,6 +162,7 @@ func TestWatchEntireDirectory(t *testing.T) {
 	}
 
 	target.Stop()
+	positions.Stop()
 
 	buf, err := ioutil.ReadFile(filepath.Clean(positionsFileName))
 	if err != nil {
@@ -269,6 +271,7 @@ func TestFileRolls(t *testing.T) {
 	}
 
 	target.Stop()
+	positions.Stop()
 
 	if len(client.messages) != 20 {
 		t.Error("Handler did not receive the correct number of messages, expected 20 received", len(client.messages))
@@ -339,6 +342,7 @@ func TestResumesWhereLeftOff(t *testing.T) {
 	}
 
 	target.Stop()
+	positions.Stop()
 
 	// Create another positions (so that it loads from the previously saved positions file).
 	positions2, err := NewPositions(logger, PositionsConfig{
@@ -367,6 +371,7 @@ func TestResumesWhereLeftOff(t *testing.T) {
 	}
 
 	target2.Stop()
+	positions2.Stop()
 
 	if len(client.messages) != 20 {
 		t.Error("Handler did not receive the correct number of messages, expected 20 received", len(client.messages))
@@ -449,6 +454,7 @@ func TestGlobWithMultipleFiles(t *testing.T) {
 	}
 
 	target.Stop()
+	positions.Stop()
 
 	buf, err := ioutil.ReadFile(filepath.Clean(positionsFileName))
 	if err != nil {

